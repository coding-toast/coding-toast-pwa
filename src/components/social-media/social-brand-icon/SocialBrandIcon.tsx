import * as React from 'react';
import { Col, Image } from 'react-bootstrap';
import Link from 'next/link';
import { ISocialBrand } from '../types';

interface ISocialBrandIconProps {
  brand: ISocialBrand;
  size: number;
}

const SocialBrandIcon: React.FC<ISocialBrandIconProps> = (props) => {
  const { brand, size } = props;
  return (
    <Col className='mx-auto text-center' xs='auto'>
      <Link href={brand.href}>
        <Image src={brand.logo} alt={brand.name} style={{ height: `${size}em` }} />
      </Link>
    </Col>
  );
};

export default SocialBrandIcon;
