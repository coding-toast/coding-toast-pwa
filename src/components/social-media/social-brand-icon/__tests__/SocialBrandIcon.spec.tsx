import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SocialBrandIcon from '../SocialBrandIcon';

describe('<SocialBrandIcon />', () => {
  it('renders correctly', () => {
    const brand = {
      href: '/',
      logo: '/Twitter.svg',
      name: 'Twitter'
    };
    const component = shallow(<SocialBrandIcon brand={brand} size={2} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
