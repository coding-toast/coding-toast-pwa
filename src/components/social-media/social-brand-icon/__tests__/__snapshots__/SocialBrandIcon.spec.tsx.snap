// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialBrandIcon /> renders correctly 1`] = `
<Col
  className="mx-auto text-center"
  xs="auto"
>
  <Link
    href="/"
  >
    <Image
      alt="Twitter"
      fluid={false}
      rounded={false}
      roundedCircle={false}
      src="/Twitter.svg"
      style={
        Object {
          "height": "2em",
        }
      }
      thumbnail={false}
    />
  </Link>
</Col>
`;
